src/sentry/analytics/event.py

```diff
@@ -9,23 +9,43 @@
 
 
 class Attribute(object):
-    __slots__ = ['name', 'type', 'required']
-
     def __init__(self, name, type=six.text_type, required=True):
         self.name = name
         self.type = type
         self.required = required
 
+    def serialize(self, value):
```

Comment:
I'm not sure of a name offhand ("extract" might be a better verb but this has more responsibilities than just extracting values from the input that at this point, incl. type validation).

The "serialization" part here is really just converting the input type to a different output type and *preparing* it for serialization, but serialization is [more or less the act of converting a data structure into a byte stream](https://en.wikipedia.org/wiki/Serialization) that can be transmitted/stored serially. (Our API "serializer" isn't serialization either.)